### PR TITLE
V1 Unit Tests for Groups

### PR DESCRIPTION
**Pull request by andrewhand**

### CODE REVIEW

File: tests/unit/restapi/v1/test_groups.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+        "lastModifiedOn",
+    }
+    assert set(response.keys()) == expected_keys
+
+    # Validate the non-Ref fields
+    assert isinstance(response["id"], int)
+    assert isinstance(response["name"], str)
+    assert isinstance(response["createdOn"], str)
+    assert isinstance(response["lastModifiedOn"], str)
+    assert isinstance(response["latestSnapshot"], bool)
```
Comment (keithmanville) [inline tests/unit/restapi/v1/test_groups.py:94]: group doesn't have snapshots.

File: tests/unit/restapi/v1/test_groups.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+            does not match the expected response or if the response contents is not
+            valid.
+    """
+    expected_keys = {
+        "id",
+        "name",
+        "user",
+        "createdOn",
+        "lastModifiedOn",
+    }
```
Comment (keithmanville) [inline tests/unit/restapi/v1/test_groups.py:86]: missing the `members` field.

File: tests/unit/restapi/v1/test_groups.py [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+    - The user registers a group named "new_group".
+    - The response is valid matches the expected values given the registration request.
+    - The user is able to retrieve information about the group using the group id^[1].
+
+    [1] The group id is generated in the backend during the POST and is returned in the response.
+    """
+    name = "new_group"
+    user_id = auth_account["user_id"]
+
```
Comment (keithmanville) [inline tests/unit/restapi/v1/test_groups.py:247]: nitpick: remove whitespace

File: tests/unit/restapi/v1/test_groups.py [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+    - The user retrieves information about the same group and it reflects the name
+      change.
+    - The user issues a request to change the name of a group to an existing group's
+      name.
+    - The request fails with an appropriate error message and response code.
+    """
+    updated_group_name = "updated_name"
+    group_to_rename = registered_groups["group1"]
+    existing_group = registered_groups["group2"]
+
```
Comment (keithmanville) [inline tests/unit/restapi/v1/test_groups.py:358]: nitpick: reduce whitespace

**Review by keithmanville — CHANGES_REQUESTED** [review]

File: tests/unit/restapi/v1/test_groups.py (hunk 2 above)
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py:87]: Added in commit: [d06c575](https://github.com/usnistgov/dioptra/pull/458/commits/d06c5758eeedc1c5771d2c4ded820b8f4b212ab6)

File: tests/unit/restapi/v1/test_groups.py (hunk 3 above)
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py]: Removed in commit: [0d7f47c](https://github.com/usnistgov/dioptra/pull/458/commits/0d7f47c2b08459889a5064a463670fa441bfcca1)

File: tests/unit/restapi/v1/test_groups.py (hunk 4 above)
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py]: Removed in commit: [5e972db](https://github.com/usnistgov/dioptra/pull/458/commits/5e972db226019468a4c54429528b4863cfd5366d)

**Review by jkglasbrenner — CHANGES_REQUESTED** [review]
In addition to the inline comments, we also mentioned during this morning's session that we've added a description field for all of the resources in Dioptra, so that's something that can be incorporated in this pull request.

File: tests/unit/restapi/v1/test_groups.py [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+    auth_account: dict[str, Any],
+    registered_groups: dict[str, Any],
+) -> None:
+    """Test that groups can be queried with a search term.
+
+    Given an authenticated user and registered groups, this test validates the following
+    sequence of actions:
+
+    - The user is able to retrieve a list of all registered groups with a description
+      that contains 'group'.
```
[the remark this answers is not in the document]
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py:298]: Added in commit: [016ba98](https://github.com/usnistgov/dioptra/pull/458/commits/016ba982181530cc6a2c8e3010a1cc29832c9f43)

File: tests/unit/restapi/v1/test_groups.py [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+        # Validate the GroupRef structure for member
+        assert isinstance(member["group"]["id"], int)
+        assert isinstance(member["group"]["name"], str)
+
+        # Validate permissions
+        assert isinstance(member["read"], bool)
+        assert isinstance(member["write"], bool)
+        assert isinstance(member["share_read"], bool)
+        assert isinstance(member["share_right"], bool)
+
```
[the remark this answers is not in the document]
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py:126]: Added in commit: [016ba98](https://github.com/usnistgov/dioptra/pull/458/commits/016ba982181530cc6a2c8e3010a1cc29832c9f43)

File: tests/unit/restapi/v1/test_groups.py [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+    sequence of actions:
+
+    - The user is able to retrieve single group by its ID.
+    - The response is a single group with a matching ID.
+    """
+    group3_expected = registered_groups["group3"]
+
+    assert_retrieving_group_by_id_works(
+        client, group_id=group3_expected["groupId"], expected=group3_expected
+    )
```
[the remark this answers is not in the document]
Comment (andrewhand) [reply tests/unit/restapi/v1/test_groups.py]: Removed in commit" [dd4e120](https://github.com/usnistgov/dioptra/pull/458/commits/dd4e120e3e8f7e75bf3015dbb25b8861dcdf29f5)

**Comment by andrewhand** [comment]
> 2b90d9e

Added descriptions to groups testing in commit: [2b90d9e](https://github.com/usnistgov/dioptra/pull/458/commits/2b90d9e0900f37b6c067f5de20ba13ca83710864)

Will need a review before merge @jkglasbrenner @keithmanville

**Comment by keithmanville** [comment]
integrated into #468

**Pull request closed without merging**